todo-app/src/App.js

```diff
@@ -1,10 +1,86 @@
 import React, { Component } from 'react';
+import Header from './Components/Header'
+import Footer from './Components/Footer'
+import Container from './Components/Container'
+
+
+import './App.css';
+const data = require('./todoList.json');
 
 class App extends Component {
+  state = {
+    todoItems: []
+  }
+  componentWillMount() {
+    localStorage.getItem('todoItems') && this.setState({
+      todoItems: JSON.parse(localStorage.getItem('todoItems'))
+    })
+  }
+  componentDidMount() {
+    if (!localStorage.getItem('todoItems')) {
+      this.setState({
+        todoItems: data
+      })
+    } else {
+      console.log('we are using the data from the Local Storage.')
+    }
+
+  }
+  componentWillUpdate(nextProps, nextState) {
+    localStorage.setItem('todoItems', JSON.stringify(nextState.todoItems))
+  }
+
+  handelSelected = (index) => {
+    const todoItems = JSON.parse(localStorage.getItem('todoItems'))
```

Comment:
This is wrong. You should be trusting React to handle state, not handling it yourself via localStorage. If you want to update your state, you should reference `this.state` to get current values. If you are using localStorage to manage state on your own, you are not using React in the intended way.